internal/repository/sync.go

```diff
@@ -26,9 +26,13 @@ var (
 func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Repository, error) {
 	if dir == "" {
 		// FIXME: my understanding is backslashes are not used in URLs but I haven't dived into the RFCs to check yet
-		split := strings.Split(strings.TrimSpace(strings.ReplaceAll(remoteURL, "\\", "/")), "/")
+		modifiedURL := strings.ReplaceAll(remoteURL, "\\", "/")
+		modifiedURL = strings.TrimRight(strings.TrimSpace(modifiedURL), "/") // Trim space and trailing slashes if present
+
+		split := strings.Split(modifiedURL, "/")
 		dir = strings.TrimSuffix(split[len(split)-1], ".git")
```

Comment:
1. Can you trim the `/` _specifically_ for the `dir` rather than the URL?
2. Can you add a test with a trailing `/` and see if it passes?

I ask because I also want us to see if go-git's behavior matches git's. I think it does, the issue you opened failed at the directory creation.

Reply:
I am not sure if I understand correctly. ModifiedURL is only used to create split. The original remoteURL does not get modified.

Reply:
> I ask because I also want us to see if go-git's behavior matches git's. I think it does, the issue you opened failed at the directory creation.

I just checked and it does not in some cases. If there are two trailing slashes go-git is unable to clone, while git in-fact does.

Reply:
> I am not sure if I understand correctly. ModifiedURL is only used to create split. The original remoteURL does not get modified.

Ah I see. I missed that, my bad. That should be okay then, I suggest we pass in the URL as-is to gitinterface for the most part.

Reply:
> I just checked and it does not in some cases. If there are two trailing slashes go-git is unable to clone, while git in-fact does.

That sounds like a fairly straightforward bug to fix in go-git :eyes: